src/Core_Fix_ResourceUnloadOptimizations/Core.ResourceUnloadOptimizations.cs

```diff
@@ -16,19 +18,38 @@ public partial class ResourceUnloadOptimizations
         private static Func<AsyncOperation> _originalUnload;
 
         private static int _garbageCollect;
+        private static readonly List<Func<bool>> IsLoadingChecks = new List<Func<bool>>
+        {
+            GetIsNowLoadingFade
+        };
+
+#if !EC
```

Comment:
All of the `#if !EC` directives made the code into a bit of a swiss cheese, would it be possible to reduce them or at least consolidate them?

Reply:
Some of these could be just left out and the code could always be there but you may get` "unused XXX" messages. If that's preferable I'll do that

Reply:
Updated: I added `InsideStudio => false` for EC Constants instead, and everything should just work.